api/krusty/longlinebreak_test.go

```diff
@@ -0,0 +1,44 @@
+package krusty_test
+
+import (
+	"testing"
+
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+)
+
+func TestLongLineBreaks(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteF("/app/deployment.yaml", `
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: test
+  labels:
+    app.version: "{{ .Values.services.mariadb.app.version }}"
+spec:
+  template:
+    spec:
+      containers:
+      - name: mariadb
+        image: "thisIsAReallyLongRepositoryLinkThatResultsInALineBreakWhenBuildingWithKustomize/mariadb:{{ .Values.services.mariadb.image.version }}"
+`)
+	th.WriteK("/app", `
+resources:
+- deployment.yaml
+`)
+	m := th.Run("/app", th.MakeDefaultOptions())
```

Comment:
Can you add a comment here, e.g.

// The problem is the linebreak that appears at the end of
// the `image:` line field before the closing quote.

I was staring at this wondering what the problem was :)

Reply:
actually no - the quotes are dropped.

Reply:
Actually multiple things are getting mixed up here.
This is making less sense.

Reply:
OK, i think the immediate goal is to preserve quotes on string fields that contain blanks.
The line length doesn't matter here, does it?
If so, the test could use a short string, e.g. 
```
image: "a b"
```
The PR title is confusing me.

Reply:
Unless i'm mistaken after a few minutes of searching, blanks aren't allowed in image names, and we generally aren't (cannot) worry about supporting moustache or Go template notation in string fields (that's not how one should combine helm and kustomize).

Let's get a test in place that shows the behavior of fields containing string values that are long and quoted and contain blanks.

Suggest using the example in https://github.com/kubernetes-sigs/kustomize/issues/947#issuecomment-638316334

which has the value in an `env` field rather than an image field.

But for the value just use the following:
```
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
```
i.e. don't use template syntax.  just a long, quoted string

Reply:
Hi @monopole, I stumbled across this PR and your comment "that's not how one should combine helm and kustomize". I'm curious, how would you recommend integrating Kustomize with a templating tool like Helm?

We need to add dynamic information generated by Terraform to our Kubernetes configs. Thanks!